src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -665,6 +665,11 @@ void OTARequestor::RecordNewUpdateState(OTAUpdateStateEnum newState, OTAChangeRe
         // Inform the driver that the OTARequestor has entered the Idle state
         mOtaRequestorDriver->HandleIdleState(idleStateReason);
     }
+    else if (((mCurrentUpdateState == OTAUpdateStateEnum::kIdle) || (mCurrentUpdateState == OTAUpdateStateEnum::kUnknown)) &&
```

Comment:
We basically never use `kUnknown` other then the first time booting up. I don't think you really need to check for it here.